exporter/elasticexporter/internal/translator/elastic/exceptions.go

```diff
@@ -0,0 +1,204 @@
+	current := Exception{out, nil, 0}
+	stack := []Exception{}
+	scanner := bufio.NewScanner(strings.NewReader(s))
+	for scanner.Scan() {
+		if first {
+			// Ignore the first line, we only care about the locations.
+			first = false
+			continue
+		}
+		var indent int
+		line := scanner.Text()
+		if i := strings.IndexFunc(line, isNotTab); i > 0 {
+			line = line[i:]
+			indent = i
+		}
+		for indent < current.indent {
+			n := len(stack)
+			current, stack = stack[n-1], stack[:n-1]
+		}
+		switch {
+		case strings.HasPrefix(line, "at "):
+			if err := parseJavaStacktraceFrame(line, current.Exception); err != nil {
+				return err
+			}
+		case strings.HasPrefix(line, "..."):
+			// "... N more" lines indicate that the last N frames from the enclosing
+			// exception's stacktrace are common to this exception.
+			if current.enclosing == nil {
+				return fmt.Errorf("no enclosing exception preceding line %q", line)
+			}
+			submatch := javaStacktraceMoreRegexp.FindStringSubmatch(line)
+			if submatch == nil {
+				return fmt.Errorf("failed to parse stacktrace line %q", line)
+			}
+			if n, err := strconv.Atoi(submatch[1]); err == nil {
+				enclosing := current.enclosing
+				if len(enclosing.Stacktrace) < n {
+					return fmt.Errorf(
+						"enclosing exception stacktrace has %d frames, cannot satisfy %q",
+						len(enclosing.Stacktrace), line,
+					)
+				}
+				m := len(enclosing.Stacktrace)
+				current.Stacktrace = append(current.Stacktrace, enclosing.Stacktrace[m-n:]...)
+			}
+		case strings.HasPrefix(line, causedByPrefix):
+			// "Caused by:" lines are at the same level of indentation
+			// as the enclosing exception.
+			current.Cause = make([]model.Exception, 1)
+			current.enclosing = current.Exception
+			current.Exception = &current.Cause[0]
+			current.Exception.Handled = current.enclosing.Handled
+			current.Message = line[len(causedByPrefix):]
+		case strings.HasPrefix(line, suppressedPrefix):
+			// Suppressed exceptions have no place in the Elastic APM
+			// model, so they are ignored.
+			//
+			// Unlike "Caused by:", "Suppressed:" lines are indented within their
+			// enclosing exception; we just account for the indentation here.
+			stack = append(stack, current)
+			current.enclosing = current.Exception
+			current.Exception = &model.Exception{}
+			current.indent = indent
+		default:
+			return fmt.Errorf("unexpected line %q", line)
+		}
+	}
+	return scanner.Err()
+}
+
+func parseJavaStacktraceFrame(s string, out *model.Exception) error {
+	submatch := javaStacktraceAtRegexp.FindStringSubmatch(s)
+	if submatch == nil {
+		return fmt.Errorf("failed to parse stacktrace line %q", s)
+	}
+	var module string
+	function := submatch[1]
+	if slash := strings.IndexRune(function, '/'); slash >= 0 {
+		// We could have either:
+		//  - "class_loader/module/class.method"
+		//  - "module/class.method"
+		module, function = function[:slash], function[slash+1:]
+		if slash := strings.IndexRune(function, '/'); slash >= 0 {
```

Comment:
Does the regex make sure there's no slash at the end of the function? Either way, probably safe to have a bounds check since we definitely don't want to panic.